packages/vitest/src/node/coverage.ts

```diff
@@ -130,12 +131,18 @@ export class BaseCoverageProvider<Options extends ResolvedCoverageOptions<'istan
       this.options.reportsDirectory,
       tempDirectory,
     )
+
+    this.roots = ctx.config.project?.length
+      ? ctx.projects.map(project => project.config.root)
```

Comment:
Just realized that projects can have identical `root` too (e.g. https://github.com/AriPerkkio/aria-live-capture/blob/cc0b9d87e3d8d70e2f115c6a9a1ef6bc03195e89/vitest.config.ts#L20-L40) when they are used just to test different settings. 

Let's make sure we don't hold duplicate entries in `this.roots` to avoid unnecessary globbing in other methods:

```suggestion
      ? [...new Set(ctx.projects.map(project => project.config.root))]
```